### PR TITLE
update conventional library to latest version

### PR DESCRIPTION
**Pull request by olevett**
Relates to #1.

This updates the Conventional library to the latest, and fixes tests broken in the process.

### CODE REVIEW

File: Conventional.Samples.Tests/Danger/ConventionsToPreventDangerousBehaviour.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public void DtosShouldEagerlyEvaluateEnumerables()
         {
             typeof (AttendeeInformationDto).Assembly.GetExportedTypes()
                 .Where(x => x.Namespace != null && x.Namespace.EndsWith(".Dtos"))
-                .MustConformTo(Convention.EnumerablePropertiesMustBeEagerLoadedConventionSpecification);
```
Comment (olevett) [inline Conventional.Samples.Tests/Danger/ConventionsToPreventDangerousBehaviour.cs:15]: Was renamed

File: Conventional.Samples/1 - Domain/Attendee.cs [hunk 2]
```diff
@@ -19,8 +19,8 @@ public static Attendee Create(Guid id)
             return new Attendee(id);
         }
 
-        public Guid Id { get; }
+        public Guid Id { get; private set; }
```
Comment (olevett) [inline Conventional.Samples/1 - Domain/Attendee.cs:22]: The `PropertiesMustHavePrivateSetters` convention requires this - I'm not actually sure if that's "right", given backing fields/readonly props, but that's probably not a discussion for here!

**Comment by andrewabest** [comment]
I'm going to take a look at these today @olevett, thanks for your contributions!

**Comment by andrewabest** [comment]
Thanks @olevett - sorry about the delayed feedback loop here! I very much appreciate your contributions.

**Pull request merged**